### PR TITLE
Fix when case when node-labeller doesn't have access to kvm

### PR DESCRIPTION
**Pull request by ksimon1**
**What this PR does / why we need it**:
fix when case when node-labeller doesn't have access to kvm device

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubevirt/kubevirt/issues/5364

**Special notes for your reviewer**:

**Release note**:
```
NONE
```
Signed-off-by: Karel Šimon <ksimon@redhat.com>

### CODE REVIEW

File: pkg/virt-handler/node-labeller/node_labeller.go
Comment (ormergi) [inline pkg/virt-handler/node-labeller/node_labeller.go:144]: What you think about updating `loadedCPUInfo` boolean as part of  `loadCPUInfo` function?

File: pkg/virt-handler/node-labeller/node_labeller.go
Comment (ormergi) [inline pkg/virt-handler/node-labeller/node_labeller.go:153]: What you think about updating `loadedSupportedFeatures` boolean as part of  `loadHostSupportedFeatures` function?

File: pkg/virt-handler/node-labeller/node_labeller.go
Comment (ormergi) [inline pkg/virt-handler/node-labeller/node_labeller.go:162]: What you think about updating `loadedHostCapabilities` boolean as part of `loadHostCapabilities` function?

File: pkg/virt-handler/node-labeller/node_labeller.go
Comment (ormergi) [inline pkg/virt-handler/node-labeller/node_labeller.go:166]: What you think about updating `loadedHyperv` boolean as part of `loadHypervFeatures` function?

File: pkg/virt-handler/node-labeller/node_labeller.go
Comment (ormergi) [inline pkg/virt-handler/node-labeller/node_labeller.go:63]: nit: What you think about renaming these to "XXIsLoaded",e.g: `CPUInfoIsLoaded`, `HyperVIsLoaded`?

File: pkg/virt-handler/node-labeller/node_labeller.go
Comment (ormergi) [inline pkg/virt-handler/node-labeller/node_labeller.go:64]: nit: With the current name I would expect it to be the actual error that been raised
What you think about rnameing `kvmErrorPosted` => `kvmPostedError`?

File: pkg/virt-handler/node-labeller/node_labeller.go
Comment (omeryahud) [inline pkg/virt-handler/node-labeller/node_labeller.go:112]: Is this here to prevent multiple prints because the labelling mechanism runs indefinitely?

Comment (ksimon1) [reply pkg/virt-handler/node-labeller/node_labeller.go:112]: yes exactly

**Review by ormergi — APPROVED** [review]
Thank you

File: pkg/virt-handler/node-labeller/node_labeller.go
Comment (davidvossel) [inline pkg/virt-handler/node-labeller/node_labeller.go:124]: If this fails, why would it ever pass after being retried multiple times?

if this function is not eventually consistent, it needs to only be executed once at initialization, and a decision needs to be made at that point in time about how to handle a potential error.

**Review by davidvossel — COMMENTED** [review]
how did the previous node labeller handle situations where no kvm device is present?

Does it even make sense to run the node labeller when no kvm device exists? I'm assuming this would mean emulation mode is enabled or something.

**Comment by davidvossel** [comment]
/hold

I want to make sure there's not a more efficient way of doing this before we merge. If we detect that the node labeller will never be useful for some reason at startup, and that situation has no chance of improving throughout the lifecycle of the virt-handler pid, then we should consider not running the node labeller reconcile loop when that condition is hit.

**Comment by ksimon1** [comment]
when kvm is not presented, previous version of labeller just ends and never checks if kvm is available. I am not sure if kvm can be installed during lifecycle of virt-handler?

**Comment by ksimon1** [comment]
@davidvossel I created new commit, which disables labeller, when kvm is not available. Now you can choose, which behaviour is better.

**Review by davidvossel — COMMENTED** [review]
There's a lot going on here, and I'm uncertain why.  Is there a reason why this can't be as simple as...

make `kvmPath` exported as `KvmPath` in the nodelabeller package 

then in `cmd/virt-handler/virt-handler.go`

```
deviceController := &devicemanager.DeviceController{}
if device.Controller.NodeHasDevice(nodelabeller.KvmPath)
        nodeLabellerController, err := nodelabeller.NewNodeLabeller(deviceController, app.clusterConfig, app.virtCli, app.HostOverride, app.namespace)
        if err != nil {
                panic(err)
        }
         
        go nodeLabellerController.Run(10, stop)
} else {
      n.logger.Warningf("node-labeller is disabled, cannot work without KVM device.")
}

**Comment by ksimon1** [comment]
> There's a lot going on here, and I'm uncertain why. Is there a reason why this can't be as simple as...
> 
> make `kvmPath` exported as `KvmPath` in the nodelabeller package
> 
> then in `cmd/virt-handler/virt-handler.go`
> 
> ```
> deviceController := &devicemanager.DeviceController{}
> if device.Controller.NodeHasDevice(nodelabeller.KvmPath)
>         nodeLabellerController, err := nodelabeller.NewNodeLabeller(deviceController, app.clusterConfig, app.virtCli, app.HostOverride, app.namespace)
>         if err != nil {
>                 panic(err)
>         }
>          
>         go nodeLabellerController.Run(10, stop)
> } else {
>       n.logger.Warningf("node-labeller is disabled, cannot work without KVM device.")
> }
> ```

Updated

**Review by davidvossel — APPROVED** [review]
/lgtm
/approve

**Comment by erkanerol** [comment]
@davidvossel  If you are ok with this PR, could you please remove the hold?

**Comment by ksimon1** [comment]
/hold cancel
removing hold to unblock HCO

**Comment by davidvossel** [comment]
> @davidvossel If you are ok with this PR, could you please remove the hold?

oops, sorry about that. Yes this is good to go, good judgement call @ksimon1, thanks for removing the hold for me :+1:

**Pull request merged**
